backend/src/main/scala/com/softwaremill/adopttapir/template/scala/EndpointsView.scala

```diff
@@ -173,12 +182,28 @@ object EndpointsView {
 
   }
 
+  private object MetricsEndpoint {
+    def prepareMetricsEndpoint(serverEffect: ServerEffect): Code = {
+      val (effect, endpoint) = serverEffect match {
```

Comment:
There is redundancy with doc endpoints.  EndpointsView:212.  Maybe you could extract method for this common code.